security-groups.yaml

```diff
@@ -0,0 +1,417 @@
+# Required Python packages:
+#
+# ansible
+# openstackclient
+# openstacksdk
+
+- import_playbook: common.yaml
+
+- hosts: all
+  gather_facts: no
+
+  tasks:
+  - name: 'Create the master security group'
+    os_security_group:
+      name: "{{ os_sg_master }}"
+
+  - name: 'Set master security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_master }} "
+
+  - name: 'Create the worker security group'
+    os_security_group:
+      name: "{{ os_sg_worker }}"
+
+  - name: 'Set worker security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_worker }} "
+
+  - name: 'Create the infra security group'
+    os_security_group:
+      name: "{{ os_sg_infra}}"
+
+  - name: 'Set infra security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_infra }} "
+
+  - name: 'Create the API security group'
+    os_security_group:
+      name: "{{ os_sg_api }}"
+
+  - name: 'Set API security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_api }} "
+
+  - name: 'Create the ingress security group'
+    os_security_group:
+      name: "{{ os_sg_ingress }}"
+
+  - name: 'Set ingress security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_ingress }} "
+
+  - name: 'Create the SSH security group'
+    os_security_group:
+      name: "{{ os_sg_ssh }}"
+
+  - name: 'Set SSH security group tag'
+    command:
+      cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_ssh }} "
+
+  - name: 'Create master-sg rule "ICMP"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_master }}"
+      protocol: icmp
+
+  - name: 'Create master-sg rule "machine config server"'
```

Comment:
Some of these could use with_items which would make the file shorter and the list easier to maintain. If this is working now though I'm not suggesting we change it all now.